pkg/controller/deployment/deployment_controller.go

```diff
@@ -415,6 +416,15 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		glog.V(4).Infof("Ignoring paused deployment %s/%s", d.Namespace, d.Name)
 		return nil
 	}
+	if d.Spec.RollbackTo != nil {
+		revision := d.Spec.RollbackTo.Revision
+		if err := dc.rollback(d, &revision); err != nil {
```

Comment:
I still think that you should return here and not allow the deployment to be processed anymore during this cycle.

Think about the following scenario:
1. Controller start syncing a deployment d with d.ResourceVersion = 10 and d.Spec.Rollback != nil
2. `rollback(d, &revision)` succeeds in rolling back the deployment and cleaning up the rollback config so the new updated object D has D.ResourceVersion = 11 and D.Spec.Rollback == nil.
3. The reconcilation cycle continues but you are still using d which has d.ResourceVersion = 10
4. A new rc will be created/promoted to the latest revision of the deployment and then you will try to update the annotations of d with the latest revision which will result in an update conflict due to the outdated resource version, right?

Haven't you already hit such an error? I would be surprised.